src/languageSupport/languages/flux/lsp/connection.ts

```diff
@@ -91,6 +111,193 @@ class LspConnectionManager {
     this._worker.postMessage(msg)
   }
 
+  _toggleSchemaSync() {
+    if (this._session.composition.synced) {
+      return this._callbackSetSession({...DEFAULT_SCHEMA})
+    }
+    this._callbackSetSession({
+      ...this._session,
+      composition: {...this._session.composition, synced: true},
+    })
+  }
+
+  _setCompositionSyncToggle() {
+    setTimeout(() => {
+      const syncIcon = document.getElementsByClassName(ICON_SYNC_ON)[0]
+      const clickableInvisibleDiv = document.getElementById(ICON_SYNC_ID)
```

Comment:
Right now, the icon is placed within the monaco-editor by passing classnames to the editor, and asking the classname be applied (per line in the code block). The icon is added to line 1. But this icon is not clickable.

In order to create a clickable element with a handler, we place an invisible div over the icon. In this way we get the best of both worlds:
* the monaco-editor itself places the icon on the right line, so it always gets the proper top+left position regardless of responsive re-sizing.
* Then we use that exact top+left position --> to then set the invisible clickable div exactly correct.